packages/realm-web/src/App.ts

```diff
@@ -173,7 +173,15 @@ export class App<
             () => this.deviceInformation,
         );
         // Hydrate the app state from storage
-        this.hydrate();
+        try {
+            this.hydrate();
+        } catch (err) {
+            // The storage was corrupted
+            this.storage.clear();
+            // A failed hydration shouldn't throw and break the app experience
+            // Since this is "just" persisted state that unfortunately got corrupted or partially lost
+            console.warn("Realm app hydration failed:", err.message);
```

Comment:
This is the major change, we're logging a warning instead of throwing an error (that the user cannot do anything about anyway).

Reply:
Is the `this.storage.clear()` call needed? It seems like the corrupted value is remove already [AppStorage.getUserIds()](https://github.com/realm/realm-js/blob/master/packages/realm-web/src/AppStorage.ts#L41-L57).

Reply:
That only clears the list of user ids, not the storage related to the entire app.

Reply:
I believe it is, this is more broad as it will handle any corruption to the storage, including errors thrown while dehydrating a user instance: https://github.com/realm/realm-js/blob/master/packages/realm-web/src/User.ts#L134-L142. The error I was specifically trying to address is the `providerType` property which was added recently and users authenticated with a previous version of the SDK won't be able to be dehydrated from storage.

I could however, most likely remove the `catch` and `this.remove(USER_IDS_STORAGE_KEY);` from `AppStorage.getUserIds()` 🤔

Reply:
I removed the `this.remove` from `AppStorage.getUserIds`.